components/purchase_modal/purchase_modal.tsx

```diff
@@ -252,11 +264,32 @@ export default class PurchaseModal extends React.PureComponent<Props, State> {
                     </a>
                 </div>
                 <div className='central-panel'>
-                    <PaymentForm
-                        className='normal-text'
-                        onInputChange={this.onPaymentInput}
-                        onCardInputChange={this.handleCardInputChange}
-                    />
+                    {this.state.editPaymentInfo ?
+                        <PaymentForm
+                            className='normal-text'
+                            onInputChange={this.onPaymentInput}
+                            onCardInputChange={this.handleCardInputChange}
+                        /> :
+                        <div className='paymentInfoDetails'>
```

Comment:
Should we be checking here if they have payment details in the state to show in saved payment mode?